Source/Core/Core/Boot/Boot_BS2Emu.cpp

```diff
@@ -242,6 +242,30 @@ bool CBoot::EmulatedBS2_GC(const DiscIO::VolumeDisc& volume)
   return RunApploader(/*is_wii*/ false, volume);
 }
 
+static DiscIO::Region CodeRegion(char c)
+{
+  switch (c)
+  {
+  case 'J':  // Japan
+  case 'T':  // Taiwan
+    return DiscIO::Region::NTSC_J;
+  case 'B':  // Brazil
+  case 'M':  // Middle East
+  case 'R':  // Argentina
+  case 'S':  // ???
+  case 'U':  // USA
+  case 'W':  // ???
+    return DiscIO::Region::NTSC_U;
+  case 'A':  // Australia
+  case 'E':  // Europe
+    return DiscIO::Region::PAL;
+  case 'K':  // Korea
+    return DiscIO::Region::NTSC_K;
+  default:
+    return DiscIO::Region::Unknown;
+  }
+}
+
 bool CBoot::SetupWiiMemory(IOS::HLE::IOSC::ConsoleType console_type)
 {
   static const std::map<DiscIO::Region, const RegionSetting> region_settings = {
```

Comment:
Can these serial mappings be changed? "LJ" isn't a valid prefix for JPN, and "LE" isn't a valid one for PAL. Japanese and PAL serial number prefixes have three chars. I'd suggest LJH and LEH (so it's identical to KOR for which it is LKH).